src/hotspot/share/opto/convertnode.cpp

```diff
@@ -238,11 +236,14 @@ Node *ConvF2LNode::Ideal(PhaseGVN *phase, bool can_reshape) {
 //------------------------------Value------------------------------------------
 const Type* ConvHF2FNode::Value(PhaseGVN* phase) const {
   const Type *t = phase->type( in(1) );
-  if( t == Type::TOP ) return Type::TOP;
-  if( t == TypeInt::SHORT ) return Type::FLOAT;
-  const TypeInt *ti = t->is_int();
-  if ( ti->is_con() ) return TypeF::make( SharedRuntime::hf2f( ti->get_con() ) );
+  if (t == Type::TOP) return Type::TOP;
+  if (t == TypeInt::SHORT) return Type::FLOAT;
+  if (StubRoutines::hf2f() == nullptr) return bottom_type();
 
+  const TypeInt *ti = t->is_int();
+  if (ti->is_con()) {
```

Comment:
I find it confusing that `ConvHF2FNode::Value()` has `is_con()` check, but `ConvF2HFNode::Value()`doesn't. I'd prefer to see both implementations unified.